stdlib/http/src/main/java/org/ballerinalang/net/http/ClientListener.java

```diff
@@ -0,0 +1,86 @@
+
+import java.io.IOException;
+
+import static org.ballerinalang.net.http.WebSocketConstants.STATEMENT_FOR_RECONNECT;
+import static org.ballerinalang.net.http.WebSocketConstants.STATUS_CODE_ABNORMAL_CLOSURE;
+import static org.ballerinalang.net.http.WebSocketUtil.doReconnect;
+import static org.ballerinalang.net.http.WebSocketUtil.hasRetryConfig;
+import static org.ballerinalang.net.http.WebSocketUtil.setCloseMessage;
+
+/**
+ * Client connector listener for WebSocket.
+ */
+public class ClientListener extends WebSocketClientConnectorListener {
+    private WebSocketOpenConnectionInfo connectionInfo;
+    private static final Logger logger = LoggerFactory.getLogger(ClientListener.class);
+
+    public void setConnectionInfo(WebSocketOpenConnectionInfo connectionInfo) {
+        this.connectionInfo = connectionInfo;
+    }
+
+    @Override
+    public void onMessage(WebSocketCloseMessage webSocketCloseMessage) {
+        ObjectValue webSocketClient = connectionInfo.getWebSocketEndpoint();
+        int statusCode = webSocketCloseMessage.getCloseCode();
+        if (hasRetryConfig(webSocketClient) && statusCode == STATUS_CODE_ABNORMAL_CLOSURE) {
+            if (!doReconnect(connectionInfo)) {
+                logger.info(STATEMENT_FOR_RECONNECT +
+                        webSocketClient.getStringValue(WebSocketConstants.CLIENT_URL_CONFIG));
+                setCloseMessage(connectionInfo, webSocketCloseMessage);
+            }
+        } else {
+            if (hasRetryConfig(webSocketClient) && statusCode != STATUS_CODE_ABNORMAL_CLOSURE) {
+                logger.info("Couldn't connect to the server because the given server: " +
+                        webSocketClient.getStringValue(WebSocketConstants.CLIENT_URL_CONFIG) +
+                        "have sent the close request to the client.");
+            }
+            setCloseMessage(connectionInfo, webSocketCloseMessage);
+        }
+    }
+
+    @Override
+    public void onError(WebSocketConnection webSocketConnection, Throwable throwable) {
+        ObjectValue webSocketClient = connectionInfo.getWebSocketEndpoint();
+        if (throwable instanceof IOException) {
+            if (hasRetryConfig(webSocketClient)) {
+                if (!doReconnect(connectionInfo)) {
+                    logger.info(STATEMENT_FOR_RECONNECT +
+                            webSocketClient.getStringValue(WebSocketConstants.CLIENT_URL_CONFIG));
+                }
+            }
+            WebSocketDispatcher.dispatchError(connectionInfo, throwable);
+        } else {
+            if (hasRetryConfig(webSocketClient)) {
+                logger.info("Unable do the retry because the connection: " +
+                        webSocketClient.getStringValue(WebSocketConstants.CLIENT_URL_CONFIG) +
+                        "has some issue that needs to fix.");
+            }
+            WebSocketDispatcher.dispatchError(connectionInfo, throwable);
```

Comment:
I think we can clean up these `if-else` blocks as well.
   ```ballerina
     if (throwable instanceof IOException && hasRetryConfig(webSocketClient) && doReconnect(connectionInfo)) {
            return;
        } else if (hasRetryConfig(webSocketClient)) {
            logger.info("Unable do the retry because the connection: " +
                    webSocketClient.getStringValue(WebSocketConstants.CLIENT_URL_CONFIG) +
                    "has some issue that needs to fix.");
        }
    WebSocketDispatcher.dispatchError(connectionInfo, throwable);
```
WDYT?